### PR TITLE
feat: add Zeabur deployment

### PR DESCRIPTION
**Pull request by yuaanlin**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added documentation and preset for [Zeabur](https://zeabur.com).

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by pan93412 — COMMENTED** [review]
LGTM! Any demo site of this?

**Comment by yuaanlin** [comment]
> LGTM! Any demo site of this?

Sure, here is a Nuxt.js app deployed on Zeabur: https://nuxtjs.zeabur.app/ (source: https://github.com/zeabur/nuxtjs-template).

And the `zeabur` preset in this PR is already tested:

```
❯ NITRO_PRESET=zeabur pnpm nitro build examples/hello-world/

> nitropack@2.8.0 nitro /Users/yuanlin/Developer/nitro
> NODE_OPTIONS="--enable-source-maps" jiti ./src/cli/index.ts "build" "examples/hello-world/"

✔ Generated public examples/hello-world/.zeabur/output/static                                                                                         nitro 2:32:11 PM
ℹ Building Nitro Server (preset: zeabur)                                                                                                              nitro 2:32:11 PM
✔ Nitro server built                                                                                                                                  nitro 2:32:11 PM
  ├─ examples/hello-world/.zeabur/output/functions/__nitro.func/chunks/index.mjs (267 B) (191 B gzip)
  ├─ examples/hello-world/.zeabur/output/functions/__nitro.func/chunks/index.mjs.map (170 B) (136 B gzip)
  ├─ examples/hello-world/.zeabur/output/functions/__nitro.func/chunks/nitro/zeabur.mjs (140 kB) (33.9 kB gzip)
  ├─ examples/hello-world/.zeabur/output/functions/__nitro.func/chunks/nitro/zeabur.mjs.map (3.7 kB) (751 B gzip)
  ├─ examples/hello-world/.zeabur/output/functions/__nitro.func/index.mjs (224 B) (174 B gzip)
  └─ examples/hello-world/.zeabur/output/functions/__nitro.func/package.json (113 B) (108 B gzip)
Σ Total size: 145 kB (35.2 kB gzip)
```

Screenshot 2023-11-22 at 2 41 01 PM

```
nitro on  main [!] is 📦 v2.8.0 via  v20.8.0 on ☁️  (ap-northeast-1) 
❯ NITRO_PRESET=zeabur pnpm nitro build examples/api-routes/

> nitropack@2.8.0 nitro /Users/yuanlin/Developer/nitro
> NODE_OPTIONS="--enable-source-maps" jiti ./src/cli/index.ts "build" "examples/api-routes/"

✔ Generated public examples/api-routes/.zeabur/output/static                                                                                          nitro 2:33:00 PM
ℹ Building Nitro Server (preset: zeabur)                                                                                                              nitro 2:33:00 PM
✔ Nitro server built                                                                                                                                  nitro 2:33:00 PM
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/_..._.mjs (433 B) (257 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/_..._.mjs.map (234 B) (152 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/_name_.mjs (288 B) (200 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/_name_.mjs.map (249 B) (163 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/hello.mjs (258 B) (185 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/hello.mjs.map (167 B) (132 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/nitro/zeabur.mjs (141 kB) (34 kB gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/nitro/zeabur.mjs.map (3.7 kB) (751 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/test.get.mjs (266 B) (185 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/test.get.mjs.map (174 B) (133 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/test.post.mjs (270 B) (187 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/chunks/test.post.mjs.map (176 B) (133 B gzip)
  ├─ examples/api-routes/.zeabur/output/functions/__nitro.func/index.mjs (224 B) (174 B gzip)
  └─ examples/api-routes/.zeabur/output/functions/__nitro.func/package.json (113 B) (108 B gzip)
Σ Total size: 147 kB (36.7 kB gzip)
```

Screenshot 2023-11-22 at 2 39 40 PM

File: docs/content/2.deploy/20.providers/zeabur.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Zeabur
+
+Deploy Nitro apps to [Zeabur](https://zeabur.com).
+
+**Preset:** `zeabur` ([switch to this preset](/deploy/#changing-the-deployment-preset))
+
+::alert
+**Zero Config Provider**
+:br
+Integration with this provider is possible with zero configuration. ([Learn More](/deploy/#zero-config-providers))
```
Comment (pi0) [inline docs/content/2.deploy/20.providers/zeabur.md:10]: This is amazing! But to be a zero-config provider, I expect the platform's CI/CD to expose a known environment variable which we can detect with [std-env](https://github.com/unjs/std-env) and also nitro to auto switch the preset.

File: docs/content/2.deploy/20.providers/zeabur.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+::
+
+## Deploy using Git
+
+1. Push your code to your git repository (Currently only GitHub supported).
+2. [Import your project](https://zeabur.com/docs/get-started) into Zeabur.
+3. Zeabur will detect that you are using Nitro and will enable the correct settings for your deployment.
```
Comment (pi0) [inline docs/content/2.deploy/20.providers/zeabur.md:17]: How is it possible today?

File: docs/content/2.deploy/20.providers/zeabur.md (hunk 1 above)
Comment (pi0) [reply docs/content/2.deploy/20.providers/zeabur.md:10]: Thanks! Feel free to open a PR for std-env also once ready 👍🏼

File: docs/content/2.deploy/20.providers/zeabur.md (hunk 2 above)
Comment (yuaanlin) [reply docs/content/2.deploy/20.providers/zeabur.md:17]: Zeabur has an open-sourced project [zbpack](https://github.com/zeabur/zbpack) to detect the project configuration, including programing language, framework and other configs.

Currently, Zeabur will detect a Nuxt.js project and auto inject the `NITRO_PRESET` environment variable to build the project with proper command (https://github.com/zeabur/zbpack/blob/main/pkg/zeaburpack/main.go#L213-L221). But I think we need a stable method to determine a project is build with Nitro (even if it's not Nuxt.js project).

Comment (yuaanlin) [reply docs/content/2.deploy/20.providers/zeabur.md:17]: @pi0 I suddenly realized something: after this Pull Request is merged, all frameworks developed based on Nitro (whether they're Nuxt.js or not) will automatically use `NITRO_PRESET=zeabur` when building, determined by the `ZEABUR=1` environment variable. The output from the completed build will then comply with the Zeabur format, eliminating the need for additional conversions through zbpack as required by current Nuxt.js projects!

**Review by pi0 — APPROVED** [review]
Let's try on nightly!

**Pull request merged**